homeassistant/components/velbus/__init__.py

```diff
@@ -19,61 +21,91 @@
     })
 }, extra=vol.ALLOW_EXTRA)
 
+COMPONENT_TYPES = ['switch', 'sensor', 'binary_sensor', 'cover', 'climate']
+
 
 async def async_setup(hass, config):
     """Set up the Velbus platform."""
-    import velbus
+    # Import from the configuration file if needed
+    if DOMAIN not in config:
+        return True
+
     port = config[DOMAIN].get(CONF_PORT)
-    controller = velbus.Controller(port)
 
-    hass.data[DOMAIN] = controller
+    if not port:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={'source': SOURCE_IMPORT}))
+    else:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={'source': SOURCE_IMPORT},
+                data={
```

Comment:
Just have the `if` check logic define the data to pass. The call can be moved outside the check.